### PR TITLE
Removed the asset price, changed constants

### PR DESCRIPTION
**Pull request by Andrei-Masilevich**
1. SP is traded to the SCR one to one
2. Increased coin precision

### CODE REVIEW

File: tests/tests/basic_tests.cpp [hunk 1]
```diff
@@ -26,10 +26,50 @@
...
 
 BOOST_AUTO_TEST_SUITE(type_operation_tests)
 
+BOOST_AUTO_TEST_CASE(check_asset_limits)
+{
+    BOOST_CHECK_GT(SCORUM_MAX_SHARE_SUPPLY, 0);
+
+    BOOST_CHECK_LE(SCORUM_MAX_SHARE_SUPPLY, std::numeric_limits<share_value_type>::max());
+
+    BOOST_CHECK_GE(asset::maximum(SCORUM_SYMBOL).amount, SCORUM_MAX_SHARE_SUPPLY);
```
Comment (abolinbsu) [inline tests/tests/basic_tests.cpp]: change to EQUAL

File: libraries/protocol/include/scorum/protocol/config.hpp [hunk 2]
```diff
@@ -116,7 +116,9 @@
 #define SCORUM_MAX_VOTE_CHANGES                5
 #define SCORUM_REVERSE_AUCTION_WINDOW_SECONDS  (60*30) /// 30 minutes
 #define SCORUM_MIN_VOTE_INTERVAL_SEC           3
-#define SCORUM_VOTE_DUST_THRESHOLD             (50)
+#define SCORUM_VOTE_DUST_THRESHOLD             share_value_type(50)
+
+#define SCORUM_MAX_SHARE_SUPPLY                share_value_type(99999999e+9) //100 million - 1
```
Comment (abolinbsu) [inline libraries/protocol/include/scorum/protocol/config.hpp]: change to 1000...

File: libraries/protocol/include/scorum/protocol/scorum_operations.hpp [hunk 3]
```diff
@@ -167,7 +167,7 @@ struct comment_options_operation : public base_operation
     std::string permlink;
 
     asset max_accepted_payout
```
Comment (abolinbsu) [inline libraries/protocol/include/scorum/protocol/scorum_operations.hpp]: I would remove max_accepted_payout. The only perpose it's used is  checking max value in util::get_rshare_reward()

File: tests/tests/operation_tests.cpp [hunk 4]
```diff
@@ -3220,16 +2849,6 @@ BOOST_AUTO_TEST_CASE(escrow_release_validate)
         op.agent = "sam";
         op.receiver = "bob";
 
-        BOOST_TEST_MESSAGE("--- failure when scorum < 0");
```
Comment (alexplevako) [inline tests/tests/operation_tests.cpp]: why you delete this?

Comment (Andrei-Masilevich) [reply tests/tests/operation_tests.cpp]: ok, this test will be restored. But I will fix asset class to receive only SCR or SP symbols.

File: libraries/protocol/include/scorum/protocol/scorum_operations.hpp (hunk 3 above)
Comment (Andrei-Masilevich) [reply libraries/protocol/include/scorum/protocol/scorum_operations.hpp]: This is used for business logic. Look get_rshare_reward

**Pull request merged**